icepyx/core/query.py

```diff
@@ -104,11 +104,14 @@ def __init__(
         self, spatial_extent=None, date_range=None, start_time=None, end_time=None
     ):
         # validate & init spatial extent
-        self.extent_type, self._spat_extent, self._geom_filepath = val.spatial(
-            spatial_extent
-        )
+
+        sp_extent = sp.Spatial(spatial_extent)
+        self.extent_type = sp_extent.extent_type
+        self._spat_extent = sp_extent.spatial_extent
+        self._geom_filepath = sp_extent.extent_file
```

Comment:
since we're turning spatial into a class, it would be cool if we just attached the spatial object to the GenQuery object. So we might have something like `self.spatial = sp.Spatial(spatial_extent)` and then would access the needed info throughout `query.py` as `self.spatial.extent_type`, rather than having to manually set the variables here. Happy to work through variable names and which should be private (starting with `_`) with you.

Reply:
Resolved with recent pushes to the branch